biofefi/pages/2_Data_Preprocessing.py

```diff
@@ -0,0 +1,325 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import streamlit as st
+from sklearn.feature_selection import VarianceThreshold
+from sklearn.linear_model import Lasso
+from sklearn.preprocessing import MinMaxScaler, StandardScaler
+
+from biofefi.components.experiments import experiment_selector
+from biofefi.components.images.logos import sidebar_logo
+from biofefi.options.choices import NORMALISATIONS, TRANSFORMATIONS_Y
+from biofefi.options.enums import ConfigStateKeys, Normalisations, TransformationsY
+from biofefi.options.file_paths import (
+    biofefi_experiments_base_dir,
+    data_preprocessing_options_path,
+    execution_options_path,
+    plot_options_path,
+    raw_data_path,
+)
+from biofefi.options.preprocessing import PreprocessingOptions
+from biofefi.services.configuration import (
+    load_execution_options,
+    load_plot_options,
+    save_options,
+)
+from biofefi.services.experiments import get_experiments
+
+
+def build_config() -> PreprocessingOptions:
+    """
+    Build the configuration object for preprocessing.
+    """
+
+    preprocessing_options = PreprocessingOptions(
+        feature_selection_methods={
+            ConfigStateKeys.VarianceThreshold: st.session_state[
+                ConfigStateKeys.VarianceThreshold
+            ],
+            ConfigStateKeys.CorrelationThreshold: st.session_state[
+                ConfigStateKeys.CorrelationThreshold
+            ],
+            ConfigStateKeys.LassoFeatureSelection: st.session_state[
+                ConfigStateKeys.LassoFeatureSelection
+            ],
+        },
+        dependent_variable_normalisation=st.session_state[
+            ConfigStateKeys.DependentNormalisation
+        ].lower(),
+        independent_variable_transformation=st.session_state[
+            ConfigStateKeys.IndependentNormalisation
+        ].lower(),
+    )
+    return preprocessing_options
+
+
+def run_feature_selection(feature_selection_methods: dict, data: pd.DataFrame) -> None:
```

Comment:
1. I don't like the fact that this reassigning `data` within this function. I think you should return build a new dataframe and return that instead.
2. Docstring
3. Could this function go into a module under the services folder please?